src/main/java/frc/robot/joystickManager/JoystickBindings.java

```diff
@@ -0,0 +1,22 @@
+package frc.robot.joystickManager;
+
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import frc.robot.Robot;
+import frc.joysticks.SmartJoystick;
+
+public class JoystickBindings {
+
+	public static void configureBindings(SmartJoystick joystick, Robot robot) {
+		applySwerveBindnigs(joystick, robot);
+		applySecondBindings(joystick, robot);
+	}
+
+	private static void applySwerveBindnigs(SmartJoystick joystick, Robot robot) {
+		joystick.A.and(() -> joystick.getState() == JoystickState.SWERVE).whileTrue(new RunCommand(() -> System.out.println("swerve")));
```

Comment:
how would swerve type commands that need to hold the joystick work with this?

Reply:
also if this pattern of ".and(() -> joystick.getState() == somestate)" is going to appear repeatedly then it should be enforced by some kind of wrapper or something like that such that it will only allow to add a binding with a state attached to it

Reply:
it was only for testing purposes, thank you i will remove those lines,
and about the second comment, what if you want to apply a binging that will be active in all the states?

Reply:
I dont think that is ever a requirement
if there is some sort of weird edge case you can just apply it to all state one by one but it really seems unlikely